vspec/utils/vssidval.py

```diff
@@ -0,0 +1,449 @@
+        """Validates uid and either assigns next higher id or overwrites with validation id.
+        If used in automatic mode it will always assign a new higher id for mismatches. In
+        manual mode it will ask you what you would like to do.
+
+        Args:
+            k (str): current key in signals dict
+            v (dict): current value contains another dict
+            match_tuple (tuple): is the name and id of the matched names in validation tree
+        """
+        nonlocal highest_id
+        nonlocal assigned_new_uid
+        validation_node: VSSNode = validation_tree_nodes[match_tuple[1]]
+
+        try:
+            assert v["staticUID"] == validation_node.extended_attributes["staticUID"]
+
+        except AssertionError:
+            if assigned_new_uid:
+                logging.warning(
+                    f"UID MISMATCH also, but new static UID for '{k}' has already "
+                    "been assigned."
+                )
+            else:
+                logging.warning(
+                    "[Validation] "
+                    f"UID MISMATCH: IDs don't match. Current tree's node '{key}' has "
+                    f"static UID '{v['staticUID']}' and validation tree's node "
+                    f"'{validation_node.qualified_name()}' has static UID "
+                    f"'{validation_node.extended_attributes['staticUID']}'!"
+                )
+                if config.validate_automatic_mode:
+                    assign_new_id(key, value)
+                else:
+                    user_interaction(key, value, match_tup=match_tuple)
+                assigned_new_uid = True
+
+    def check_unit(k: str, v: dict, match_tuple: tuple):
+        """Validates if the unit of a node has changed in comparison to the validation file
+
+        Args:
+            k (str): current key in signals dict
+            v (dict): current value contains another dict
+            match_tuple (tuple): is the name and id of the matched names in validation tree
+        """
+        nonlocal config
+        nonlocal highest_id
+        nonlocal assigned_new_uid
+
+        validation_node: VSSNode = validation_tree_nodes[match_tuple[1]]
+
+        if "unit" in v.keys() and "unit" in validation_node.source_dict.keys():
+            try:
+                assert v["unit"] == validation_node.source_dict["unit"]
+            except AssertionError:
+                if assigned_new_uid:
+                    logging.warning(
+                        f"UNIT MISMATCH also, but a new id for '{k}' has already "
+                        "been assigned."
+                    )
+                else:
+                    logging.warning(
+                        "[Validation] "
+                        f"UNIT MISMATCH: Units of '{k}' in unit: '{v['unit']}' "
+                        f"in the current specification and the validation file "
+                        f"'{k}' in unit: '{validation_node.unit}' don't "
```

Comment:
Using `f"'{k}'`means that the new name is written, not the old, see other comment